core/eth/contracts.go

```diff
@@ -63,8 +63,8 @@ func GetV6ContractCodec(name string) (ContractCodec, error) {
 	return getContractCodec(name, box)
 }
 
-func (contract *contractCodec) ABI() abi.ABI {
-	return contract.abi
+func (contract *contractCodec) ABI() *abi.ABI {
+	return &contract.abi
 }
```

Comment:
Should this return a copy of the ABI to try and prevent altering internal structure, or what that the point>